internal/server/util/http.go

```diff
@@ -288,3 +291,67 @@ func IsJSONRequest(r *http.Request) bool {
 
 	return false
 }
+
+// CheckJwtToken checks whether the given request has JWT token that is valid and
+// signed with client certificate from the trusted certificates.
+// Returns whether or not the token is valid, the fingerprint of the certificate and the certificate.
+func CheckJwtToken(r *http.Request, trustedCerts map[string]x509.Certificate) (bool, string, *x509.Certificate) {
+	// Check if the request has a JWT token
+	auth := r.Header.Get("Authorization")
+	if auth == "" {
+		return false, "", nil
+	}
+	parts := strings.Split(auth, " ")
+	if len(parts) != 2 || strings.ToLower(parts[0]) != "bearer" {
+		return false, "", nil
+	}
+
+	tokenString := parts[1]
+
+	// Parse the token
+	jwtParser := jwt.NewParser()
+
+	requestToken, requestTokenParts, err := jwtParser.ParseUnverified(tokenString, &jwt.RegisteredClaims{})
+	if err != nil {
+		return false, "", nil
+	}
+
+	requestTokenClaims, ok := requestToken.Claims.(*jwt.RegisteredClaims)
+	if !ok {
+		return false, "", nil // JWT claims type invalid
+	}
+
+	// Check if the token is valid
+	if time.Now().Before(requestTokenClaims.NotBefore.Time) || time.Now().After(requestTokenClaims.ExpiresAt.Time) {
+		return false, "", nil // token not yet valid or expired
+	}
+
+	// Find the certificate by token subject
+	requestTokenCert, ok := trustedCerts[requestTokenClaims.Subject]
+	if !ok {
+		return false, "", nil // certificate not found
+	}
```

Comment:
We need to validate that the certificate is still valid at this stage since we don't have the TLS handler doing that for us here.

Reply:
The certificate validation is in the cmd/incusd/daemon.go:470
We can move it here if more appropriate.

Reply:
Oh, sorry I missed that. It's fine to do it from daemon.go.